packages/pwa-kit-extension-sdk/src/configs/webpack/overrides-resolver-plugin.ts

```diff
@@ -76,7 +72,12 @@ export class OverridesResolverPlugin {
                 extensions: this.options.fileExtensions,
                 packageIterator: () =>
                     buildCandidatePaths(importPath, sourcePath, {
-                        extensionEntries: this.options.extensions,
+                        // TODO: does the overrides resolver need to get the _configured_ extensions at build time?
+                        // If yes, then it looks like we'll need a convention of having at least default.json config,
+                        // where ALL configured extensions are listed there, and have them all enabled.
+
+                        // Ensure we have the long form configuration entry.
+                        extensionEntries: expand(getConfig()?.app?.extensions),
```

Comment:
Overrides and their current implementation is by definition something that happens at build time. I don't think there is going to be any way around that. Thinking quickly off the top of my head, if we wanted to have some sort of runtime override concept, it would involve lots of chunks of code and dynamic imports. 

I have a short conversation with John and what we'll end up doing for those partners that want to have different extension setups for different environments is that we'll have to manage that on their own via build management. 

So, does that invalidate the work that we have going now? maybe, maybe not. If there is a world where overrides goes bye bye, the it could be beneficial to have this solution of determining extensions at runtime vs at build time. But for the time being I think this is going to be a known issue that we release our beta with. E.g. overrides doesn't work at runtime and configuration X will be used to define overrieds at build time.